ngraph/frontend/onnx_import/src/op/dropout.cpp

```diff
@@ -0,0 +1,136 @@
+// limitations under the License.
+//*****************************************************************************
+
+#include <memory>
+
+#include "core/null_node.hpp"
+#include "default_opset.hpp"
+#include "exceptions.hpp"
+#include "ngraph/log.hpp"
+#include "ngraph/node.hpp"
+#include "op/dropout.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace
+            {
+                OutputVector
+                    build_dropout(const Node& node, float drop_probability, bool training_mode)
+                {
+                    CHECK_VALID_NODE(
+                        node,
+                        drop_probability == 0 || !training_mode,
+                        "Training mode is not supported for Dropout op if drop_probability is not "
+                        "equal 0");
+
+                    const auto input_data = node.get_ng_inputs().at(0);
+                    const bool return_mask = node.get_outputs_size() > 1;
+
+                    if (return_mask)
+                    {
+                        NGRAPH_WARN << "Default mask for Dropout is ignored, "
+                                    << "because of unsupported constant networks";
+                        /*const auto mask = std::make_shared<default_opset::Broadcast>(
+                                default_opset::Constant::create(ngraph::element::boolean,
+                            Shape{}, {true}),
+                                std::make_shared<default_opset::ShapeOf>(input_data));*/
+                        // If constant network is supported mask should be returned instead of
+                        // NullNode (ticket 48055)
+                        return {input_data, std::make_shared<NullNode>()};
+                    }
+                    else
+                    {
+                        return {input_data};
+                    }
+                }
+            }
+
+            namespace set_12
+            {
+                OutputVector dropout(const Node& node)
+                {
+                    const auto ng_inputs = node.get_ng_inputs();
+                    // seed attribute is ignored because traning mode is not supported anyway
+
+                    // default values of inputs
+                    double ratio = 0.5f;
+                    bool training_mode = false;
+
+                    if (ng_inputs.size() > 1)
+                    {
+                        if (!ngraph::op::is_null(ng_inputs.at(1)))
+                        {
+                            CHECK_VALID_NODE(
+                                node,
+                                ngraph::op::is_constant(ng_inputs.at(1).get_node_shared_ptr()),
+                                "Not constant (or omitted) ratio input is not supported.");
+                            ratio = as_type_ptr<default_opset::Constant>(
+                                        ng_inputs.at(1).get_node_shared_ptr())
```

Comment:
We shouldn't require Constant ratio as it's not needed during inference.
Also as we can read in the [onnx Dropout](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Dropout) doc, 
we can just ignore this `ratio` input if there is no `training_mode` input or it is set to`false`:

>_training_mode_ (optional, non-differentiable) : 
If set to true then it indicates dropout is being used for training. It is an optional value hence **unless specified explicitly, it is false**. **If it is false, ratio is ignored** and the operation mimics inference mode where nothing will be dropped from the input data and if mask is requested as output it will contain all ones.

Reply:
good point ;)